cmd/admin-bucket-handlers.go

```diff
@@ -164,7 +164,12 @@ func (a adminAPIHandlers) SetRemoteTargetHandler(w http.ResponseWriter, r *http.
 		writeErrorResponseJSON(ctx, w, errorCodes.ToAPIErrWithErr(ErrAdminConfigBadJSON, err), r.URL)
 		return
 	}
-	host, port, err := net.SplitHostPort(target.Endpoint)
+	turl := target.URL()
+	endPoint := turl.Host
+	if turl.Port() == "" {
+		endPoint = turl.Host + ":"
+	}
+	host, port, err := net.SplitHostPort(endPoint)
```

Comment:
This is wrong @poornas if we know port is empty why are we even doing net.SplitHostPort()? and if we already know the port turl.Port() why even do net.SplitHostPort?

Reply:
Note: you can use 

port := url.Port()
host := url.Hostname() (this is without port)

url.Host although has both port (if set) and host

Reply:
net.SplitHostPort is not necessary since the info is already present in target url, removing this.